burn-book/src/import/onnx-model.md

```diff
@@ -1,137 +1,210 @@
 use burn_import::onnx::ModelGen;
 
 fn main() {
-    // Generate Rust code from the ONNX model file
     ModelGen::new()
-        .input("src/model/mnist.onnx")
+        .input("src/model/my_model.onnx")
         .out_dir("model/")
         .run_from_script();
 }
 ```
 
-#### Step 2: Modify `mod.rs`
+This script uses `ModelGen` to generate Rust code from your ONNX model during the build process.
 
-Add this code to the `mod.rs` file located in `src/model`:
+### Step 2: Modify `mod.rs`
 
-```rust, ignore
-pub mod mnist {
-    include!(concat!(env!("OUT_DIR"), "/model/mnist.rs"));
+In your `src/model/mod.rs` file, include the generated code:
+
+```rust
+pub mod my_model {
+    include!(concat!(env!("OUT_DIR"), "/model/my_model.rs"));
 }
 ```
 
-#### Step 3: Utilize Imported Model
+This makes the generated model code available in your project.
 
-Here's how to use the imported model in your application:
+### Step 3: Use the Imported Model
 
-```rust, ignore
-mod model;
+Now you can use the imported model in your Rust code:
 
+```rust
 use burn::tensor;
 use burn_ndarray::{NdArray, NdArrayDevice};
-use model::mnist::Model;
+use model::my_model::Model;
 
 fn main() {
-    // Initialize a new model instance
     let device = NdArrayDevice::default();
+
+    // Create model instance and load weights from target dir (see more options below)
     let model: Model<NdArray<f32>> = Model::new(&device);
 
-    // Create a sample input tensor (zeros for demonstration)
-    let input = tensor::Tensor::<NdArray<f32>, 4>::zeros([1, 1, 28, 28], &device);
+    // Create input tensor (replace with your actual input)
+    let input = tensor::Tensor::<NdArray<f32>, 4>::zeros([1, 3, 224, 224], &device);
 
     // Perform inference
     let output = model.forward(input);
 
-    // Print the output
-    println!("{:?}", output);
+    println!("Model output: {:?}", output);
 }
 ```
 
-### Working Examples
+## Advanced Configuration
+
+The `ModelGen` struct offers several configuration options:
+
+```rust
+ModelGen::new()
+    .input("path/to/model.onnx")
+    .out_dir("model/")
+    .record_type(RecordType::NamedMpk)
+    .half_precision(false)
+    .embed_states(false)
+    .run_from_script();
+```
+
+- `record_type`: Specifies the format for storing weights (Bincode, NamedMpk, NamedMpkGz, or
+  PrettyJson).
+- `half_precision`: Use half-precision (f16) for weights to reduce model size.
+- `embed_states`: Embed model weights directly in the generated Rust code. Note: This requires
+  record type `Bincode`.
+
+## Loading and Using Models
+
+Depending on your configuration, you can load models in different ways:
+
+```rust
+// Load from a file (must copy weights from target output directory)
+let model = Model::<Backend>::from_file("path/to/weights", &device);
+
+// Load from embedded weights (if embed_states was true)
+let model = Model::<Backend>::from_embedded();
+
+// Load from the default location (useful for testing)
+let model = Model::<Backend>::new(&device);
```

Comment:
Without any code change, right now the `new(device)` method doesn't load the weights from the default location right? That's still done via the `default()` method 🤔 while `new` only initializes the module (so if you don't call `load_record` it will be randomly initialized).

Reply:
Thanks for pointing the problem. I fixed it.